jrpcfs/api.go

```diff
@@ -165,6 +165,23 @@ import (
 type MountIDAsByteArray [16]byte
 type MountIDAsString string
 
+// BypassReply contains response to BypassReq
+type BypassReply struct {
+	// TODO - update all error vs result vs, etc
+	// want to include Result and Error???
+	ID     uint64 `json:"id"`
+	Result []byte `json:"result"`
+	Error  string `json:"error"`
+}
+
+// BypassReq tunnels file system RPCs from pfsagentd to proxyfsd
+type BypassReq struct {
+	JSONrpc string `json:"jsonrpc"`
+	Method  string `json:"method"`
+	ID      uint64 `json:"id"`
+	Params  []byte
```

Comment:
So I see that you just put a byte slice here... I guess that means you plan on un-marshaling the payload of Params separately? That's just fine... though I didn't do it that way. You should be able to see how I did it... I basically un-marshaled just the things I needed at this point (in your case, you need Method only I should think)... then, once you know the Method, you'd know how to un-marshal what you find in Params []byte. That's fine, too...

As for the line, though, you'll still need to "adorn" it with `json:"params"` since that's what the field will be named "on the wire" (i.e. in JSON-RPC format).